src/platform/cc13x2_26x2/OTAImageProcessorImpl.cpp

```diff
@@ -367,8 +400,25 @@ void OTAImageProcessorImpl::HandleProcessBlock(intptr_t context)
         return;
     }
 
-    // TODO: Process block header if any
+    /* Save the fixed size header */
+    if (imageProcessor->mParams.downloadedBytes < sizeof(imageProcessor->mFixedOtaHeader))
+    {
+        uint8_t * header = reinterpret_cast<uint8_t *>(&(imageProcessor->mFixedOtaHeader));
+        if (imageProcessor->mBlock.size() + imageProcessor->mParams.downloadedBytes < sizeof(imageProcessor->mFixedOtaHeader))
+        {
+            memcpy(header + imageProcessor->mParams.downloadedBytes, imageProcessor->mBlock.data(), imageProcessor->mBlock.size());
+        }
+        else
+        {
+            memcpy(header + imageProcessor->mParams.downloadedBytes, imageProcessor->mBlock.data(),
+                   sizeof(imageProcessor->mFixedOtaHeader) - imageProcessor->mParams.downloadedBytes);
+        }
+    }
```

Comment:
Are you doing this to remove the header from the block? If so, why not just use `OTAImageHeaderParser::AccumulateAndDecode` to do it for you?

Reply:
Only the fixed header is useful for this platform. It is used to calculate the start of the payload for the device's bootloader. The `OTAImageHeaderParser` doesn't have an API to get the offset of the payload for further processing. I could have added that to the class to allow access to the `mHeaderTlvSize` or to calculate the start of the payload. But that includes a lot of extra code to store and process the TLVs that are not used by this platform.

Reply:
I wonder if the right thing to do is to split the API (or at least have the option to do either or both) such that you can just get the fixed part without parsing the rest of the TLVs. Not for this PR but would be good to have down the road.